client/state/inline-help/actions.js

```diff
@@ -22,6 +23,21 @@ import {
 import getContextualHelpResults from 'state/inline-help/selectors/get-contextual-help-results';
 import 'state/inline-help/init';
 
+/**
+ * Set the search query in the state tree for the inline help.
+ *
+ * @param {string} searchQuery - query string to persist.
+ * @returns {Function}            Action thunk.
+ */
+export function setInlineHelpSearchQuery( searchQuery = '' ) {
```

Comment:
There's no need to wrap this in a thunk. Let's return a simple action object instead. Thunks are harder to inspect and cannot be serialized.

```jsx
return {
		type: INLINE_HELP_SET_SEARCH_QUERY,
		searchQuery,
};
```

Reply:
Note that `requestInlineHelpSearchResults` can also be rewritten to avoid thunks in favor of data-layer handling, but that's a larger refactor that's optional to tackle.